db/migrate/20191214171048_add_github_to_author.rb

```diff
@@ -0,0 +1,7 @@
+class AddGithubToAuthor < ActiveRecord::Migration[5.2]
+  def change
+    add_column :authors, :github_id, :integer
+    add_column :authors, :from_github, :boolean, default: false
```

Comment:
How would you change this field to make it more extensible so we can change to a different platform or even add a new platform to work together with Github? (I'm just looking for answers, don't need to implement anything)

Reply:
I would change this field to a polymorphic one, so it allows you to use another platforms to work with Github as well as change the integration to a new one.
But this change allows you to continue using just 1 platform per author, if you need to have more than 1 integration per author, I would create a new table of integrations to belongs to author. ( I prefer to not do this one because you have to guarantee consistency between the integrations, when a change is triggered in one of the integrations, the application should change the other one. )

Reply:
Good answer =)